### DISCUSSION TITLE
Bug in javascript navigation

### DISCUSSION THREAD

**Discussion by breaman** [IntelliTect/EssentialCSharp.Web#20] [Feedback and Ideas] [answered]
When navigating around the site, I noticed that when I clicked on one of the url's on the left, the hashtag starts out with the right value but then switches to NaN after the page loads. The bug appears to be in the following piece of code where the .hash value is changed. My guess is that value doesn't really need to be set, but wasn't sure what the original idea was behind changing it in the first place so am creating an issue instead of a PR. In the below code, "target" is a string, so things are failing when we try to set the window.location.hash to be "target - 40" since we are subtracting 40 from a string value. Having the hash value change to NaN after the page loads would break any opportunity to send someone the url they are looking at and have the page jump to the right place.

```
jQuery(document).ready(function(){
    var target = window.location.hash;

    if ( target != '' ){
        var $target = jQuery(target); 
        jQuery('html, body').stop().animate({
        'scrollTop': $target.offset().top - 80},
        900, 
        'swing',function () {
        window.location.hash = target - 40 ;
        });
    }
});
```

**Comment by BenjaminMichaelis** [comment, marked as answer]
Thanks for the feedback @breaman!  This fix should be pushed to our production site now, thank you for bringing it up!
